runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodeRocksOverlay.java

```diff
@@ -0,0 +1,141 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.motherlode;
+
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.GameObject;
+import net.runelite.api.Perspective;
+import net.runelite.api.Player;
+import net.runelite.api.Region;
+import net.runelite.api.Tile;
+import net.runelite.api.WallObject;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayUtil;
+
+import javax.inject.Inject;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.awt.Polygon;
+import java.util.stream.IntStream;
+
+@Slf4j
+class MotherlodeRocksOverlay extends Overlay
+{
+	private final Client client;
+	private final MotherlodeConfig config;
+
+	private static final int REGION_SIZE = 104;
+	private static final int MAX_DISTANCE = 2350;
+
+	private static final int[] mineSpots = { 26661, 26662, 26663, 26664 };
+
+	private static final int[] rockObstacles = { 26680, 26679 };
+
+	@Inject
+	MotherlodeRocksOverlay(Client client, MotherlodeConfig config)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.UNDER_WIDGETS);
+		this.client = client;
+		this.config = config;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics, Point parent)
+	{
+		if (!config.enabled() || !config.showRocks())
+		{
+			return null;
+		}
+
+		Player local = client.getLocalPlayer();
+
+		//Render mining spot
+		renderTiles(graphics, local);
+
+		return null;
+	}
+
+	private void renderTiles(Graphics2D graphics, Player local)
+	{
+		Region region = client.getRegion();
+		Tile[][][] tiles = region.getTiles();
+
+		int z = client.getPlane();
+
+		for (int x = 0; x < REGION_SIZE; ++x)
+		{
+			for (int y = 0; y < REGION_SIZE; ++y)
+			{
+				Tile tile = tiles[z][x][y];
+
+				if (tile == null)
+				{
+					continue;
+				}
+				if (local.getLocalLocation().distanceTo(tile.getLocalLocation()) <= MAX_DISTANCE)
+				{
+					renderMine(graphics, tile);
+				}
+			}
+		}
+	}
+
+	private void renderMine(Graphics2D graphics, Tile tile)
+	{
+		//Draw the Pay-dirt spots
+		WallObject wallObject = tile.getWallObject();
+		if (wallObject != null && IntStream.of(mineSpots).anyMatch(x -> x == wallObject.getId()))
```

Comment:
Using list#contains (and making mineSpots list instead of array) would be a lot faster than creating streams on the fly and running anyMatch on them